src/interfaces/components/turbine_input.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "interfaces/components/beam_input.hpp"
+#include "types.hpp"
+
+namespace openturbine::interfaces::components {
+
+/**
+ * @brief Complete input specification for a turbine
+ *
+ * Defines the input configuration for a turbine including blades, tower, hub, and nacelle.
+ */
+struct TurbineInput {
+    /// @brief Configuration for the blades
+    std::vector<BeamInput> blades;
+
+    /// @brief Configuration for the tower
+    BeamInput tower;
+
+    /// @brief Yaw bearing inertia matrix (6x6)
+    /// includes yaw bearing and nacelle mass with inertia about yaw bearing
+    std::array<std::array<double, 6>, 6> yaw_bearing_inertia_matrix{
+        {{0., 0., 0., 0., 0., 0.},
+         {0., 0., 0., 0., 0., 0.},
+         {0., 0., 0., 0., 0., 0.},
+         {0., 0., 0., 0., 0., 0.},
+         {0., 0., 0., 0., 0., 0.},
+         {0., 0., 0., 0., 0., 1.}}
+    };
```

Comment:
This could be more compact I think

```
    std::array<std::array<double, 6>, 6> yaw_bearing_inertia_matrix{
        {{}, {}, {}, {}, {}, {0., 0., 0., 0., 0., 1.}}
    };
```